### PR TITLE
fix cmake build error when install under virtualenv

### PR DESCRIPTION
**Pull request by shinny-chengzhi**
CMake's ```FindPythonLibs``` is not working very well under virtualenv. We need to set ```PYTHON_LIBRARY``` and ```PYTHON_INCLUDE_DIR``` to help CMake locate the relevant directories.

### CODE REVIEW

**Review by timkpaine — CHANGES_REQUESTED** [review]
This won't work for manylinux builds, and this stuff is handled by the FindPython cmake modules.

**Comment by shinny-chengzhi** [comment]
It seems like the coding style check failed, and I'm not familiar with Lint Python. Could you shine any light on the error ```BLK100: Black would make changes```?

I think the culprit is the FindPythonLibs module which can't find include&lib location in my case. Here is my setup:
+ compare python form source and install it to /home/cz/work/python-3.6
+ create virtualenv under /home/cz/work/py36env
+ run ```pip3 install perspective-python``` and encountered error:
```
...
  running build_ext
  -- The C compiler identification is GNU 6.3.0
  -- The CXX compiler identification is GNU 6.3.0
  -- Check for working C compiler: /usr/bin/cc
  -- Check for working C compiler: /usr/bin/cc -- works
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Check for working CXX compiler: /usr/bin/c++
  -- Check for working CXX compiler: /usr/bin/c++ -- works
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Found Boost: /usr/include (found version "1.67.0")
  -- Found Python: /home/cz/work/py36env/bin/python3.6 (found suitable exact version "3.6.5") found components: Interpreter Development
  -- Found PythonInterp: /home/cz/work/py36env/bin/python3.6 (found suitable exact version "3.6.5")
  CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
    Could NOT find PythonLibs (missing: PYTHON_LIBRARIES PYTHON_INCLUDE_DIRS)
    (Required is exact version "3.6")
  Call Stack (most recent call first):
    /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
    /usr/share/cmake-3.16/Modules/FindPythonLibs.cmake:310 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
    CMakeLists.txt:377 (find_package)
  
  
  -- Configuring incomplete, errors occurred!
  See also "/tmp/pip-install-ubkztp2d/perspective-python/build/temp.linux-x86_64-3.6/CMakeFiles/CMakeOutput.log".
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-ubkztp2d/perspective-python/setup.py", line 259, in <module>
...
```

Even though I can run ```python -c "import sysconfig; print(sysconfig.get_config_var('INCLUDEPY'), sysconfig.get_config_var('LIBDIR'))"``` to get ```/home/cz/work/python-3.6/bin/include/python3.6m /home/cz/work/python-3.6/bin/lib``` as include&lib location.

After applying this patch, I can finally install perspective in my environment.

**Comment by shinny-chengzhi** [comment]
I tried building the latest CMake form source and the problem still preexist. Here is the log:
```
...
  running build_ext
  -- The C compiler identification is GNU 6.3.0
  -- The CXX compiler identification is GNU 6.3.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /usr/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /usr/bin/c++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Found Boost: /usr/include (found version "1.67.0")
  -- Found Python: /home/cz/work/py36env/bin/python3.6 (found suitable exact version "3.6.5") found components: Interpreter Development Development.Module Development.Embed
  -- Found PythonInterp: /home/cz/work/py36env/bin/python3.6 (found suitable exact version "3.6.5")
  CMake Error at /home/cz/work/CMake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
    Could NOT find PythonLibs (missing: PYTHON_LIBRARIES PYTHON_INCLUDE_DIRS)
    (Required is exact version "3.6")
  Call Stack (most recent call first):
    /home/cz/work/CMake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
    /home/cz/work/CMake/Modules/FindPythonLibs.cmake:310 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
    CMakeLists.txt:377 (find_package)
  
  
  -- Configuring incomplete, errors occurred!
  See also "/tmp/pip-install-qhdxkse5/perspective-python/build/temp.linux-x86_64-3.6/CMakeFiles/CMakeOutput.log".
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-qhdxkse5/perspective-python/setup.py", line 259, in <module>
...
```

It's seem like this is a known bug: https://gitlab.kitware.com/cmake/cmake/-/issues/19525

File: python/perspective/setup.py [hunk 1]
```diff
@@ -176,7 +183,7 @@ def build_extension_cmake(self, ext):
                     "-DCMAKE_TOOLCHAIN_FILE={}".format(vcpkg_toolchain_file)
                 )
 
-            if sys.maxsize > 2 ** 32:
+            if sys.maxsize > 2**32:
```
Comment (timkpaine) [inline python/perspective/setup.py:186]: needs spaces around operator for linting

**Review by timkpaine — CHANGES_REQUESTED** [review]

**Comment by shinny-chengzhi** [comment]
Strange that when I run black it's constantly striping the space for me:
```
(py36env) cz@chengzhi:~$ black work/perspective/python/perspective/setup.py 
reformatted work/perspective/python/perspective/setup.py

All done! ✨ 🍰 ✨
1 file reformatted.
```

Is there any rules that need to be fed to black? Nevertheless, I have manually added these spaces back.

**Comment by shinny-chengzhi** [comment]
I've create a new one https://github.com/finos/perspective/pull/1779 Since github didn't support changing source branch for PR.

**Pull request closed without merging**